pipeline-model-definition/src/test/resources/postStage/catchErrorStageFailureNested.groovy

```diff
@@ -0,0 +1,60 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2019, CloudBees, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+pipeline {
+    agent any
+
+    stages {
+        stage("parent") {
+            stages {
+                stage("foo") {
+                    steps {
+                        catchError(buildResult: "SUCCESS",
+                            message: "Caught an error",
+                            stageResult: "FAILURE") {
+                            error("uhoh")
+                        }
+                    }
+                    post {
+                        success {
+                            echo "This shouldn't happen"
+                        }
+                        failure {
+                            echo "This should happen"
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    post {
+        success {
+            echo "The build should be a success"
+        }
+        failure {
+            echo "The build shouldn't be a failure"
```

Comment:
I'm still confused as to why this isn't being run. I would think the relevant flow node start would be the start of the very first stage, and then `findWorstWarningBetween` would search the whole Pipeline and find the `WarningAction` in `foo`... I might pull down the PR and print out what FlowNodes are being used in the search to understand what is happening.

Reply:
This is the `post` for the *build* - so it's only looking at the overall build result, which is explicitly set to `SUCCESS`.

Reply:
Ah, gotcha, thanks for clarifying!